wpilib/wpilib/shuffleboard/component.py

```diff
@@ -0,0 +1,109 @@
+# validated: 2018-11-18 EN 175c6c1f0130 edu/wpi/first/wpilibj/shuffleboard/ShuffleboardComponent.java
+from .container import ShuffleboardContainer
+
+
+class ShuffleboardComponent:
+    def __init__(self, parent: ShuffleboardContainer, title: str, type: str=None):
+        assert parent is not None, "parent cannot be None"
+        assert title is not None, "title cannot be None"
+        self.parent = parent
+        self.title = title
+        self.type = type
+        self.metadataDirty = False
+        self.column = -1
+        self.row = -1
+        self.width = -1
+        self.height = -1
+        self.properties = {}
+
+    def getParent(self) -> ShuffleboardContainer:
+        return self.parent
+
+    def setType(self, type: str) -> None:
+        self.type = type
+        self.metadataDirty = True
+
+    def getType(self) -> str:
+        return self.type
+
+    def getTitle(self) -> str:
+        return self.title
+
+    def getProperties(self):
+        return self.properties
+
+    def withProperties(self, properties: dict) -> 'ShuffleboardComponent':
```

Comment:
Are there particular types the keys and values need to be? If so, that should also be noted in the type hint.